flink/v2.0/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java

```diff
@@ -431,4 +435,36 @@ public void testConsumeFromStartTag() throws Exception {
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("START_SNAPSHOT_ID and START_TAG cannot both be set.");
   }
+
+  @TestTemplate
+  void testWithParallelismWithProps() {
+    int customScanParallelism = defaultJobParallelism + 1;
+    sql(
+        "CREATE TABLE %s (id INT, data VARCHAR, dt VARCHAR) WITH ('scan.parallelism'='%s')",
+        TABLE, customScanParallelism);
+
+    final org.apache.flink.table.api.Table table =
+        getTableEnv().sqlQuery(String.format("select * from %s", TABLE));
+    final String explain = table.explain(ExplainDetail.JSON_EXECUTION_PLAN);
+    final String expectedPhysicalExecutionPlanFragment =
+        "\"parallelism\" : " + customScanParallelism;
+    assertThat(explain).contains(expectedPhysicalExecutionPlanFragment);
+  }
+
+  @TestTemplate
+  void testWithParallelismWithHints() {
+    sql("CREATE TABLE %s (id INT, data VARCHAR, dt VARCHAR)", TABLE);
+    int customScanParallelism = defaultJobParallelism + 1;
+
+    final org.apache.flink.table.api.Table table =
+        getTableEnv()
+            .sqlQuery(
+                String.format(
+                    "select * from %s/*+ OPTIONS('streaming'='true', 'scan.parallelism'='%s') */",
```

Comment:
are both hints and with properties passed to the table source as `properties`?

Can we also add a unit test where both are set and make sure the proper overwrite order is applied?